engine/CMakeLists.txt

```diff
@@ -208,14 +208,14 @@ IF(NOT POW_FUNCTION_EXISTS AND NOT NEED_LINKING_AGAINST_LIBM)
     # cmake_pop_check_state()
 ENDIF(NOT POW_FUNCTION_EXISTS AND NOT NEED_LINKING_AGAINST_LIBM)
 
-# Set the default build type
-IF (NOT CMAKE_BUILD_TYPE)
-    SET(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Release, RelWithDebInfo, Debug, Profiler" FORCE )
-ENDIF (NOT CMAKE_BUILD_TYPE)
-
-IF (NOT BUILD_OPT)
-    SET(BUILD_OPT -O2 CACHE STRING "-O0, -O1, -O2, -O3, -Os, -Ofast" FORCE )
-ENDIF (NOT BUILD_OPT)
+## Set the default build type
+#IF (NOT CMAKE_BUILD_TYPE)
```

Comment:
Assume these changes are temporary while you work on the branch

Reply:
@BenjamenMeyer I was actually planning to make these changes permanent. As it stands currently, the default value for `BUILD_OPT` overrides the proper compiler optimization level for the current Debug, Release, etc. configuration -- and it does so without any intervention from the user of CMake. Not ideal.

I'm also not sure that we should be setting a default build type. Especially with CMake generators that cover multiple build types at once, like the Visual Studio one. However, we can leave that discussion for later if you like.

Reply:
Our CI needs a default built type, and it's useful for most environments. I'd consider VS to be the abnormal case.

Our workflow should simply be:
```
$ mkdir build
$ cd build
$ cmake ../engine
$ make
```
And that should get you a binary. The dev shouldn't have to care about the default build type.

BTW, IIRC cmake defaults to the release type, which doesn't give us any symbols which is why we were defaulting to the RelWithDebugInfo type and overriding it.

Reply:
@BenjamenMeyer Okay. I'll leave `BUILD_OPT` out, but put the default `CMAKE_BUILD_TYPE` back in.

Reply:
Done.